test/LaunchDarkly.EventSource.Tests/EventSourceEndToEndTest.cs

```diff
@@ -29,100 +29,78 @@ public void ReceiveEventStreamInChunks()
+                    RespondWithChunks("text/event-stream", DoChunks)))
                 {
-                    expectedActions.Add(EventSink.MessageReceivedAction(new MessageEvent(MessageEvent.DefaultName, data, uri)));
-                }
-
-                var config = Configuration.Builder(uri).LogAdapter(_testLogging).Build();
-                using (var es = new EventSource(config))
-                {
-                    var sink = new EventSink(es);
-                    _ = es.StartAsync();
-                    sink.ExpectActions(expectedActions.ToArray());
+                    var expectedActions = new List<EventSink.Action>();
+                    expectedActions.Add(EventSink.OpenedAction());
+                    foreach (var data in eventData)
+                    {
+                        expectedActions.Add(EventSink.MessageReceivedAction(new MessageEvent(MessageEvent.DefaultName, data, uri)));
+                    }
+
+                    var config = Configuration.Builder(uri).LogAdapter(_testLogging).Build();
+                    using (var es = new EventSource(config))
+                    {
+                        var sink = new EventSink(es);
+                        _ = es.StartAsync();
+                        sink.ExpectActions(expectedActions.ToArray());
+                    }
                 }
             }
-        }
-
-        [Fact]
-        public void ReadTimeoutIsDetectedInDefaultStringStreamReadingMode()
-        {
-            TimeSpan readTimeout = TimeSpan.FromMilliseconds(200);
-            using (var server = StartWebServerOnAvailablePort(out var uri, ctx =>
+            finally
             {
-                ctx.Response.ContentType = "text/event-stream";
-                ctx.Response.SendChunked = true;
-                var stream = ctx.Response.OutputStream;
-                WriteChunk(stream, "data: event1\n\ndata: e");
-                Thread.Sleep(readTimeout + readTimeout);
-                WriteChunk(stream, "vent2\n\n");
-            }))
-            {
-                var config = Configuration.Builder(uri).LogAdapter(_testLogging)
-                    .ReadTimeout(readTimeout).Build();
-                using (var es = new EventSource(config))
-                {
-                    var sink = new EventSink(es) { Output = _testLogger.Debug };
-                    _ = es.StartAsync();
-                    sink.ExpectActions(
-                        EventSink.OpenedAction(),
-                        EventSink.MessageReceivedAction(new MessageEvent(MessageEvent.DefaultName, "event1", uri)),
-                        EventSink.ErrorAction(new ReadTimeoutException()),
-                        EventSink.ClosedAction()
-                        );
-                }
+                allEventsReceived.Set();
             }
         }
 
-        [Fact]
-        public void ReadTimeoutIsDetectedInRawUtf8ReadingMode()
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
```

Comment:
I replaced the two very similar `ReadTimeoutIsDetected...` test methods with a single parameterized one.